api/routes/utils.js

```diff
@@ -485,42 +484,50 @@ var utils = {
       }
 
       amount *= -1;
+
+      // Dividends pay 10%
+      var dividendsPercentage = 0.1;
+      var newInvestorReps = portfolio[index].reps + amount;
       var investment = portfolio[index].investments[j];
-      var prevAmount = investment.amount;
 
-      // Revenue is the fraction sold times the valuation
-      var revenue = Math.floor(amount / prevAmount * investment.valuation);
-      var roi = (revenue - amount)/amount;
+      var prevInvestmentAmount = investment.amount;
+      var prevInvestmentPercentage = investment.percentage;
+      var newInvestmentAmount = prevInvestmentAmount - amount;
+
+      // newPercentage / newAmount = oldPercentage / oldAmount (Proportional)
+      var newInvestmentPercentage = Math.floor((newInvestmentAmount * investment.percentage) / prevInvestmentAmount);
```

Comment:
When you revoke, the percentage shouldn't change. The percentage is the percentage of the expert you owned upon purchase.

If you decrease your investment by, half, for example, the percentage associated with the remaining half shouldn't go down. Otherwise, you lose half of your dividends by selling, and another half by the percentage cut.

The percentage doesn't actually have to do with number of reps remaining in the investment. It just has to do with the percentage at the time of investment.